biorbd_optim/path_conditions.py

```diff
@@ -134,15 +135,26 @@ def regulation(self, nb_elements):
         """
         if len(self.init) == 0:
             self.init = [0] * nb_elements
-        self.regulation_private(self.init, nb_elements, "Init")
+        if self.init_lineaire:
+            self.regulation_private(self.init, 2 * nb_elements, "Init")
```

Comment:
I think this should be 0 and not 2*nb_elements. 
first_node_init should be used as the holder for the first value and the last_node_init as the holder for the last node

Reply:
Thank you for your answer.
`self.init` is not defined here but in the line 126, `self.init = list(initial_guess)`, in an other function. By default, it is the different initial values of your state (X) or control (U). So the size of `self.init` is equal to the number of elements.
 But here, we replace the initial value by two values : the step 0 and the last one. So the size has doubled compare to before.
We can delete, replace, or modify `self.init` by itself without the first and the last node for each state or control, and so check if the size is 0, but for me it means the same and the code is more simple like this.
Also, why for line 137, if we don't take into account of the first and the last node, the size is not  `[0] * nb_elements - 2` ? I used the same logic if it is just an unique intial value.
After, we can modify the line 126, `self.init = list(initial_guess)`, by itself without the first and the last node. But it makes no sense for the case 'constant initialization'.